src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,367 @@
+                </View>
+              )
+            })}
+        </ScrollView>
+      </View>
+    )
+  }
+
+  // Display error screen no Nfts provided
+  if (nfts.length === 0) {
+    return <NftsLoadErrorScreen />
+  }
+
+  return (
+    <SafeAreaView testID="NftsInfoCarousel">
+      <PlatformSpecificTopBarButtons />
+      <ScrollView>
+        {activeNft.metadata && (
+          <FastImage
+            testID={`NftsInfoCarousel/NftImage-${activeNft.contractAddress}-${activeNft.tokenId}`}
+            style={[
+              {
+                height: scaledHeight,
+                width: variables.width,
+              },
+              isLoading && { borderRadius: 8 },
+            ]}
+            source={{
+              uri: activeNft.media.find((media) => media.raw === activeNft.metadata?.image)
+                ?.gateway,
+            }}
+            onLoad={(e) => {
+              setScaledHeight(
+                scaleImageHeight(e.nativeEvent.width, e.nativeEvent.height, variables.width)
+              )
+            }}
+            onLoadEnd={() => setIsLoading(false)}
+            onError={() => {
+              Logger.error(TAG, 'Error loading Nft image')
+            }}
+            resizeMode={FastImage.resizeMode.contain}
+          >
+            {isLoading && <MainImagePlaceholder />}
+          </FastImage>
+        )}
+        {/* This could happen if the indexer is experiencing issues with particular nfts */}
+        {!activeNft.metadata && (
+          <View style={styles.nftImageLoadingErrorContainer}>
+            <Text style={{ color: colors.light }}>{t('nftInfoCarousel.nftImageLoadError')}</Text>
+          </View>
+        )}
+
+        {nfts.filter((nft) => nft.metadata).length > 1 && <NftImageCarousel />}
+        <View style={styles.sectionContainer}>
+          <Text style={styles.title}>{activeNft.metadata?.name}</Text>
+        </View>
+        {activeNft.metadata && (
+          <View style={styles.sectionContainer}>
+            <Text style={styles.subSectionTitle}>{t('nftInfoCarousel.description')}</Text>
+            <Text style={styles.text}>{activeNft.metadata?.description}</Text>
+          </View>
+        )}
+        {activeNft.metadata?.attributes && (
+          <View style={styles.sectionContainer}>
+            <Text style={styles.subSectionTitle}>{t('nftInfoCarousel.attributes')}</Text>
+            {activeNft.metadata?.attributes.map((attribute, index) => (
+              <View key={index} style={styles.attributesContainer}>
+                <Text style={styles.attributeTitle}>{attribute.trait_type}</Text>
+                <Text style={styles.attributeText}>{attribute.value}</Text>
+              </View>
+            ))}
+          </View>
+        )}
+        {/* This should be dynamic based on Network in the future. Always show as fallback */}
+        <View style={styles.sectionContainer}>
+          <Touchable
+            onPress={() =>
```

Comment:
i think generally it's a good idea to avoid inline functions for top level components, for performance but also we'll need to add analytics later so this function will grow in complexity